### PR TITLE
Fix isCatalogMode condition and Remove the price-drop link from the the best seller display condition

### LINKED ISSUE #25277: Condition isCatalogueMode reversed + Price drop link related to the PS_DISPLAY_BEST_SELLERS condition
#### Describe the bug

- On the sitemap page, until now, the condition isCatalogueMode was reversed. 
So if you were not in catalogue mode, the best sellers + price drops page was not displayed. 
- Also, if the PS_DISPLAY_BEST_SELLERS condition is false, you cannot display the price drop page. 
Indeed, on the PricesDropController, there is no init() method which, like for the BestSalesController, checks the PS_DISPLAY_BEST_SELLERS condition.
So to standardize, they should be dissociated here

#### Expected behavior

These two links should appaar when the catalog mode is disable + price drop link must be not linked with PS_DISPLAY_BEST_SELLERS condition

#### Steps to Reproduce

1. Go to /index.php?controller=sitemap with cataloge mode disable
2. You can't see price drop + best sellers page
3. If you enable the catalog mode, go back to the sitemap, you can see the prices drop + best sellers link
4. Same if you enable or disable `PS_DISPLAY_BEST_SELLERS`configuration, prices drop link depends on it

**Screenshots**

If applicable, add screenshots or screen recordings to help explain your problem.

#### Additional information

* PrestaShop version: 1.7.6.x / 1.7.7.x
* PHP version: Any

### PR DESCRIPTION
**Pull request by Sinepel**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | On the sitemap page, until now, the condition `isCatalogueMode` was reversed. So if you were not in catalogue mode, the best sellers + price drops page was not displayed. Also, I moved the link to the price drop from the `PS_DISPLAY_BEST_SELLERS` condition.  Indeed, on the `PricesDropController`, there is no `init()` method which, as for `BestSalesController`, checks the `PS_DISPLAY_BEST_SELLERS` condition. So to standardize, I unlinked the 2 here.
| Type?             | refacto
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| Fix issue ? | Fixes #25277 
| How to test?      | On the sitemap page, the prices drop & best sellers page should appear when we are not in catalogue mode. For the best sellers, it also appears if the condition PS_DISPLAY_BEST_SELLERS is valid 
| Possible impacts? | I don't see any

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/25268)

### CODE REVIEW

**Review by Progi1984 — APPROVED** [review]

**Comment by Progi1984** [comment]
@Sinepel Could you create an issue ?

**Review by matthieu-rolland — APPROVED** [review]

**Comment by hibatallahAouadni** [comment]
FYI: this PR cannot be merged until we have PM approval (in linked issue)

**Review by PierreRambaud — APPROVED** [review]

**Comment by hibatallahAouadni** [comment]
FYI: we have the PM approval

**Comment by khouloudbelguith** [comment]
Hi @Sinepel,

Thanks for the PR.
https://github.com/PrestaShop/PrestaShop/issues/25277#issuecomment-878071247
It is ok :heavy_check_mark: 
If we disable the catalog mode, the `Best sellers` & `Price drop` are displayed.
https://drive.google.com/file/d/1gCDndGt3vcZO0uqIDdUzZaNd7S1juc99/view?usp=sharing

Thanks!

**Comment by PierreRambaud** [comment]
Thank you @Sinepel

**Pull request merged**
